### PR TITLE
bug/data_sharing

### PR DESCRIPTION
**Pull request by dephillipsmichael**
fixed crash setting user sharing scope

### CODE REVIEW

File: researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java [hunk 1]
```diff
@@ -480,19 +481,22 @@ public SharingScope getUserSharingScope() {
...
     public void setUserSharingScope(Context context, String scope) {
         StudyParticipant participant = new StudyParticipant();
+        SharingScope sharingScope = RestUtils.GSON.fromJson(scope, SharingScope.class);
+        participant.setSharingScope(sharingScope);
+
         participant.setSharingScope(SharingScope.valueOf(scope));
 
-        setUserSharingScope(scope)
+        setUserSharingScope(sharingScope)
                 .subscribe();
```
Comment (liujoshua) [inline researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java]: do you think subscribe() is enough here, or should this block until the call is done?

File: researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java [hunk 2]
```diff
@@ -480,19 +481,22 @@ public SharingScope getUserSharingScope() {
     @Override
     public void setUserSharingScope(Context context, String scope) {
         StudyParticipant participant = new StudyParticipant();
+        SharingScope sharingScope = RestUtils.GSON.fromJson(scope, SharingScope.class);
+        participant.setSharingScope(sharingScope);
+
         participant.setSharingScope(SharingScope.valueOf(scope));
```
Comment (liujoshua) [inline researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java]: Is this line supposed to be deleted?

**Review by liujoshua — COMMENTED** [review]
Thanks for fixing

File: researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java (hunk 1 above)
Comment (dephillipsmichael) [reply researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java]: I don't think it's a good idea to have a public void method in DataProvider block whatever thread it's on

File: researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java (hunk 2 above)
Comment (dephillipsmichael) [reply researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java]: Yes it is, good find.  I must have been debugging something before I pushed the code.

**Review by liujoshua — APPROVED** [review]

**Pull request merged**